pkgs/applications/science/misc/simgrid/default.nix

```diff
@@ -83,13 +83,21 @@ stdenv.mkDerivation rec {
   '';
 
   doCheck = true;
-  
+
   checkPhase = ''
     runHook preCheck
-    ctest --output-on-failure -E smpi-replay-multiple
+
+    if [ "$enableParallelBuilding" = 1 ]; then
+      if [ ''${NIX_BUILD_CORES:-0} = 0 ]; then
+        NB_CORES="$(nproc)"
+      else
+        NB_CORES="$NIX_BUILD_CORES"
+      fi
+    fi
+    ctest -j $NB_CORES --output-on-failure -E smpi-replay-multiple
```

Comment:
Doesn't -j0 work here, like with make?

Reply:
I'm not sure, nothing is specified for "0" value in the ctest documentation (https://cmake.org/cmake/help/v3.0/manual/ctest.1.html), so I did it this way.